### PR TITLE
Feature/run script improvements

### PR DESCRIPTION
**Pull request by kikkomep**
This PR fixes the problem of images not being loaded.
Missing images are caused by the `run.sh` script which is responsible for updating wikis accordingly to the tools repositories. It removes all the existing wiki resources before starting to update them. 
The new version of the script keeps available old wikis (and their related resources) while is updating them and makes available their updates only when the whole update process is completed.

### CODE REVIEW

File: bin/convert.sh [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+echo "Html folder: ${htmlFolder}"
+echo "Git Repositories file: ${gitList}"
+echo "Git branch: ${gitBranch}"
+echo -e "-------------------------------------------------------------------------------------------------------"
+
+# download the list file if it is a HTTP(s) resource
+if [[ ! -z ${gitList} && ${gitList} =~ ^https?://.+  ]]; then
+    echo -e "Downloading list of repositories..."
+    remoteGitList=${gitList}
+    gitList="/tmp/remoteGitList.txt"
```
Comment (ilveroluca) [inline bin/convert.sh]: Should be deleted at some point.  Would also be good to use `mktemp`.

File: bin/run.sh [hunk 2]
```diff
@@ -1,57 +1,65 @@
...
-      if [[ $filename =~ (README.*|\.md) ]]; then
-	 echo "Transforming $dir/$file"
-         markdown2 --extras fenced-code-blocks "$dir/$file" > "$htmlFolder/$dir/$filename"
-         cp "$htmlFolder/$dir/$filename" "$htmlFolder/$dir/$filename$extension"
-      else
-	 cp -r "$dir/$file" "$htmlFolder/$dir/$file"
-      fi
+    echo -e "\nCreating links to the updated resources..."
+    echo " - Linking new markdown folder ${newMarkdownFolder}"
+    ln -sfn ${newMarkdownFolder} ${markdownFolder}
```
Comment (ilveroluca) [inline bin/run.sh]: quotes here and below

File: bin/convert.sh [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+    else
+        cd "$(dirname "$1")"
+        echo "$(pwd -P)/$(basename "$1")"
+    fi
+}
+
+# convert md files of a git repo into html
+function convert_markdown(){
+    local container_name="${1}"
+    if [[ -d ${htmlFolder}/${container_name} ]]; then rm -Rf ${htmlFolder}/${container_name}; fi;
```
[the remark this answers is not in the document]
Comment (kikkomep) [reply bin/convert.sh]: Fixed by https://github.com/phnmnl/php-phenomenal-portal-app-library/pull/3/commits/80dbfc1679f9100979ee9b7eacc263546852d30f

File: bin/convert.sh [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+    mkdir -p "${htmlFolder}/${container_name}";
+    for file in `ls ./${container_name}`;
```
[the remark this answers is not in the document]
Comment (kikkomep) [reply bin/convert.sh]: No!

File: bin/convert.sh [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+while IFS= read line
+do
+    # skip blank lines
+    if [[ ! -z ${line} ]]; then
+        # extract the container name
+        container_name=$(echo ${line} | sed -e 's/https:\/\/\([^\/]\+\)\/\([^\/]\+\)\/\(.*\)\.git/\3/g')
+        echo -e "\nProcessing container '$container_name'..." 
+        # if the repository already exists simply update it
+        # otherwise it will be cloned
+        if [[ -d ${container_name} && ! ${forceCleanup} ]]; then
```
[the remark this answers is not in the document]
Comment (kikkomep) [reply bin/convert.sh]: Fixed by https://github.com/phnmnl/php-phenomenal-portal-app-library/pull/3/commits/80dbfc1679f9100979ee9b7eacc263546852d30f

File: bin/run.sh (hunk 2 above)
Comment (kikkomep) [reply bin/run.sh]: Fixed by https://github.com/phnmnl/php-phenomenal-portal-app-library/pull/3/commits/80dbfc1679f9100979ee9b7eacc263546852d30f

File: bin/convert.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (kikkomep) [reply bin/convert.sh]: Fixed by https://github.com/phnmnl/php-phenomenal-portal-app-library/pull/3/commits/a4bcc8a30ff738d2ceaeac72b964967639c33628

File: bin/convert.sh [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+markdownFolder="wiki-markdown"
+htmlFolder="wiki-html"
+gitList="conf/gitList.txt"
+gitBranch="master"
+targetExtension=".html"
+forceCleanup=false
+
+# parse arguments
+OTHER_OPTS=''
+while [ $# -gt 0 ]; do
```
[the remark this answers is not in the document]
Comment (kikkomep) [reply bin/convert.sh]: Fixed by https://github.com/phnmnl/php-phenomenal-portal-app-library/pull/3/commits/65be0796eeab263772b2d4eaca4f24c320926a45

File: bin/convert.sh [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+                          shift
+                          ;;
+                  * )
+                          OTHER_OPTS="$OTHER_OPTS $OPT"
+                          break
+                          ;;
+            esac
+            # Check for multiple short options
+            # NOTICE: be sure to update this pattern to match valid options
+            NEXTOPT="${OPT#-[cfr]}" # try removing single short opt
```
[the remark this answers is not in the document]
Comment (kikkomep) [reply bin/convert.sh]: Fixed by https://github.com/phnmnl/php-phenomenal-portal-app-library/pull/3/commits/65be0796eeab263772b2d4eaca4f24c320926a45

File: bin/convert.sh [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+trap handle_conversion_error ERR
+
+# process list of container repositories
+while IFS= read line
+do
+    # skip blank lines
+    if [[ ! -z ${line} ]]; then
+        # extract the container name
+        container_name=$(echo ${line} | sed -e 's/https:\/\/\([^\/]\+\)\/\([^\/]\+\)\/\(.*\)\.git/\3/g')
+        echo -e "\nProcessing container '$container_name'..."
```
[the remark this answers is not in the document]
Comment (kikkomep) [reply bin/convert.sh]: Fixed by https://github.com/phnmnl/php-phenomenal-portal-app-library/pull/3/commits/79b2aa380decbe942cb1b8c707cd728d9289d705

File: bin/convert.sh [hunk 9]
```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env bash
+
+set -o nounset
+
```
[the remark this answers is not in the document]
Comment (kikkomep) [reply bin/convert.sh]: Fixed by https://github.com/phnmnl/php-phenomenal-portal-app-library/pull/3/commits/79b2aa380decbe942cb1b8c707cd728d9289d705

File: bin/convert.sh [hunk 10]
```diff
@@ -0,0 +1,198 @@
...
+        # if the repository already exists simply update it
+        # otherwise it will be cloned
+        if [[ -d ${container_name} && ! ${forceCleanup} ]]; then
+            echo "Updating existing repository..." 
+            cd ${container_name} && git pull origin ${gitBranch} && cd ..
+        else
+            # cleanup existing git repositories is required
+            if [[ -d ${container_name} ]]; then
+                echo "Cleaning existing repositories..."
+                rm -Rf ${container_name}
```
[the remark this answers is not in the document]
Comment (kikkomep) [reply bin/convert.sh]: Fixed by https://github.com/phnmnl/php-phenomenal-portal-app-library/pull/3/commits/108845a5a9b0780d608463481dba88a1f6fc06f1

File: bin/convert.sh [hunk 11]
```diff
@@ -0,0 +1,198 @@
...
+          echo "Converting ${file} to ${filename}${targetExtension}..."
+          ${markdown2} --extras fenced-code-blocks \
+                    "${container_name}/$file" > "$htmlFolder/${container_name}/${filename}${targetExtension}"
+      fi
+    done
+}
+
+# log conversion error
+function handle_conversion_error(){
+    echo "Conversion of the container '${container_name}' failed!" >&2
```
[the remark this answers is not in the document]
Comment (kikkomep) [reply bin/convert.sh]: Fixed by https://github.com/phnmnl/php-phenomenal-portal-app-library/pull/3/commits/79b2aa380decbe942cb1b8c707cd728d9289d705 and https://github.com/phnmnl/php-phenomenal-portal-app-library/pull/3/commits/2faff892d0d1d0e8e1987588dcc472000698a971

File: bin/convert.sh (hunk 1 above)
Comment (kikkomep) [reply bin/convert.sh]: Fixed by https://github.com/phnmnl/php-phenomenal-portal-app-library/pull/3/commits/3066699f04a63699d3597bdb5db15d9d052d71f2 and https://github.com/phnmnl/php-phenomenal-portal-app-library/pull/3/commits/2faff892d0d1d0e8e1987588dcc472000698a971

File: bin/convert.sh [hunk 12]
```diff
@@ -0,0 +1,244 @@
...
+
+function log() {
+  echo -e "$(date +"%F %T") [${BASH_SOURCE}] -- $@" >&2
+}
+
+function delete_folder(){
+    local type=${1}
+    local path=${2}
+    if [[ ! -z "${path}" && -d "${path}" ]]; then
+        echo " - Removing ${type} folder ${path}"
```
Comment (ilveroluca) [inline bin/convert.sh]: use `log` function

**Pull request merged**